base/src/test/resources/success/src/List.aya

```diff
@@ -14,3 +14,48 @@ def termDesuListComp : List Nat => ([ 1, 2, 3 ]) >>= (\ a => [ 4, 5, 6 ] >>= (\
 
 def testDesuList : termList = termDesuList => idp
 def testDesuListComp : termListComp = termDesuListComp => idp
+
+def listPattern (l : List Nat) : List Nat
+  | [ ] => nil
+  | [ 1 ] => 1 :< nil
+  | [ 1, e2 ] => 1 :< e2 :< nil
+  | [ 1, e2, _ ] => l
+  | [ _, _, _, _ ] as l' => l'
+  | _ => l
+
+def testListPattern0 : listPattern nil = ([ ]) => idp
+def testListPattern1 : listPattern (1 :< nil) = ([ 1 ]) => idp
+def testListPattern2 : listPattern (1 :< 2 :< nil) = ([ 1, 2 ]) => idp
```

Comment:
Can we replace `([ ... ])` with `[ ... ]` (and also in other places) ? This can test the new parsing rule of atomic expressions.